### PR TITLE
[Go-Map] Update ProjectBuilder to generate Go Auth service

### PR DESCRIPTION
**Pull request by Lewiky**
Adds the calls into the ServerBuilder for building auth services if required by the templefile. 

Also adds all the required go snippets for just the projectbuilder unit tests to pass (they're quite large!) But they are the same as the ones already used by Will in the `GoAuthServiceGenerator` tests, so have in theory already been reviewed. 

Once this is in, I'll either update the E2E tests with a complex example, or try and update *all* of the other builders to handle auth properly

### CODE REVIEW

File: src/main/scala/temple/builder/project/ProjectConfig.scala [hunk 1]
```diff
@@ -12,6 +12,7 @@ object ProjectConfig {
   val defaultLanguage: ServiceLanguage = ServiceLanguage.Go
   val defaultDatabase: Database        = Database.Postgres
   val defaultAuth: ServiceAuth         = ServiceAuth.Email
+  val authPort: Int                    = 1023 //* One before the [[Templefile]] start point
```
Comment (c-harding) [inline src/main/scala/temple/builder/project/ProjectConfig.scala]: This looks dangerous: 1023 needs sudo, right?

Comment (c-harding) [inline src/main/scala/temple/builder/project/ProjectConfig.scala]: This looks dangerous: 1023 needs sudo, right?

Comment (c-harding) [reply src/main/scala/temple/builder/project/ProjectConfig.scala]: Also wtf is that comment symbol?

**Review by c-harding — APPROVED** [review]
So much go code. Is it identical to that on the spec repo?

File: src/main/scala/temple/builder/project/ProjectConfig.scala (hunk 1 above)
Comment (Lewiky) [reply src/main/scala/temple/builder/project/ProjectConfig.scala]: That is correct... I will update accordingly

File: src/main/scala/temple/builder/ServerBuilder.scala [hunk 2]
```diff
@@ -98,8 +98,8 @@ object ServerBuilder {
       case ServiceAuth.Email =>
         Map(
           "id"       -> Attribute(AttributeType.UUIDType),
+          "email"    -> Attribute(AttributeType.StringType()),
           "password" -> Attribute(AttributeType.StringType()),
-          "email"    -> Attribute(AttributeType.StringType(), valueAnnotations = Set(Annotation.Unique)),
```
Comment (jaylees14) [inline src/main/scala/temple/builder/ServerBuilder.scala:102]: Any reason why this is now not unique?

**Comment by willjvsmith** [comment]
Do we need a separate copy of the snippets if they're identical?

**Comment by Lewiky** [comment]
> Do we need a separate copy of the snippets if they're identical?

@smithwjv Yes, since there is exactly one difference, the package name

File: src/main/scala/temple/builder/DatabaseBuilder.scala [hunk 3]
```diff
@@ -55,6 +55,7 @@ object DatabaseBuilder {
     endpoints: Set[CRUD],
     idAttribute: IDAttribute,
```
Comment (jaylees14) [inline src/main/scala/temple/builder/DatabaseBuilder.scala]: Is this variable used now?

**Review by jaylees14 — APPROVED** [review]
Lots of Go, but changes look good :)

File: src/main/scala/temple/builder/ServerBuilder.scala (hunk 2 above)
Comment (Lewiky) [reply src/main/scala/temple/builder/ServerBuilder.scala:102]: Yep - this is a read statement so having the restriction that it's unique doesn't actually do anything ;)

File: src/main/scala/temple/builder/DatabaseBuilder.scala (hunk 3 above)
Comment (Lewiky) [reply src/main/scala/temple/builder/DatabaseBuilder.scala]: Nope - removed

**Pull request merged**